libPhoneNumber/NBGeocoderMetadataHelper.m

```diff
@@ -0,0 +1,106 @@
+                                    @"AS"
+                                    @"( "
+                                    @"SELECT 1 "
+                                    @"UNION ALL "
+                                    @"SELECT x+1 "
+                                    @"FROM   count "
+                                    @"LIMIT  length(?)), tosearch "
+                                    @"AS "
+                                    @"( "
+                                    @"SELECT substr(?, 1, x) AS indata "
+                                    @"FROM   count) "
+                                    @"SELECT   nationalnumber, "
+                                    @"description, "
+                                    @"length(nationalnumber) AS nationalnumberlength "
+                                    @"FROM     geocodingpairs%@ "
+                                    @"WHERE    nationalnumber IN tosearch "
+                                    @"ORDER BY nationalnumberlength DESC "
+                                    @"LIMIT    1";
+
+- (instancetype)initWithCountryCode:(NSNumber *)countryCode withLanguage:(NSString *)languageCode {
+  self = [super init];
+  if (self != nil) {
+    _countryCode = countryCode;
+    _language = languageCode;
+
+    NSBundle *bundle = [NSBundle bundleForClass:self.classForCoder];
+    NSURL *bundleURL = [[bundle resourceURL] URLByAppendingPathComponent:@"Resources.bundle"];
+    NSString *databasePath = [NSString stringWithFormat:@"%@%@.db", bundleURL, _language];
+    _databasePath = databasePath;
+
+    sqlite3_open([databasePath UTF8String], &_database);
+
+    sqlite3_prepare_v2(_database,
+                       [[NSString stringWithFormat:preparedStatement, countryCode] UTF8String], -1,
+                       &_selectStatement, NULL);
+  }
+  return self;
+}
+
+- (void)dealloc {
+  sqlite3_finalize(_selectStatement);
+  sqlite3_close_v2(_database);
+}
+
+- (int)createSelectStatement:(NBPhoneNumber *)phoneNumber {
+  int sqliteResultCode;
+  @autoreleasepool {
+    sqlite3_reset(_selectStatement);
+    sqliteResultCode = sqlite3_clear_bindings(_selectStatement);
+    if (sqliteResultCode == SQLITE_OK) {
+      const char *completePhoneNumber =
+          [[NSString stringWithFormat:@"%@%@", phoneNumber.countryCode, phoneNumber.nationalNumber]
+              UTF8String];
+      sqlite3_bind_text(_selectStatement, 1, completePhoneNumber, -1, SQLITE_TRANSIENT);
+      sqlite3_bind_text(_selectStatement, 2, completePhoneNumber, -1, SQLITE_TRANSIENT);
+    } else {
+      NSLog(@"The SQLite3 resulting code was: %d", sqliteResultCode);
+    }
+  }
+  return sqliteResultCode;
+}
+
+- (NSString *)searchPhoneNumber:(NBPhoneNumber *)phoneNumber {
+  if (![phoneNumber.countryCode isEqualToNumber:_countryCode]) {
+    _countryCode = phoneNumber.countryCode;
+    sqlite3_prepare_v2(_database,
+                       [[NSString stringWithFormat:preparedStatement, _countryCode] UTF8String], -1,
```

Comment:
We shouldn't create a new NSString every time here. Let's create it once and cache it somewhere

Reply:
Sounds good, I turned it into private ivar